### PR TITLE
cl21: Fix Issue 38 - Test spirv_new - new test for cl_khr_spirv_no_integer_…

### PR DESCRIPTION
**Pull request by gwawiork**
…wrap_decoration SPV_KHR_no_integer_wrap_decoration extension
fix issue https://github.com/KhronosGroup/OpenCL-CTS/issues/38

### CODE REVIEW

File: test_conformance/spirv_new/spirv_txt/ext_cl_khr_spirv_no_integer_wrap_decoration_fadd_uint.spvasm64 [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+         %15 = OpLoad %v3ulong %__spirv_BuiltInGlobalInvocationId
+         %16 = OpCompositeExtract %ulong %15 0
+         %17 = OpUConvert %uint %16
+         %18 = OpSConvert %ulong %17
+         %19 = OpInBoundsPtrAccessChain %_ptr_CrossWorkgroup_uint %lhs %18
+         %20 = OpLoad %uint %19 Aligned 4
+         %21 = OpSConvert %ulong %17
+         %22 = OpInBoundsPtrAccessChain %_ptr_CrossWorkgroup_uint %rhs %21
+         %23 = OpLoad %uint %22 Aligned 4
+         %24 = OpIAdd %uint %20 %23
```
Comment (aarongreig) [inline test_conformance/spirv_new/spirv_txt/ext_cl_khr_spirv_no_integer_wrap_decoration_fadd_uint.spvasm64]: can you change the kernel names to reflect the instruction that is being tested, in this case `fadd_uint` -> `iadd_uint`

File: test_conformance/spirv_new/spirv_txt/ext_cl_khr_spirv_no_integer_wrap_decoration_fadd_uint.spvasm64 [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+; Generator: Khronos LLVM/SPIR-V Translator; 14
+; Bound: 28
+; Schema: 0
+               OpCapability Addresses
+               OpCapability Linkage
+               OpCapability Kernel
+               OpCapability Int64
+          %1 = OpExtInstImport "OpenCL.std"
+               OpMemoryModel Physical64 OpenCL
+               OpEntryPoint Kernel %10 "fmath_cl"
```
Comment (aarongreig) [inline test_conformance/spirv_new/spirv_txt/ext_cl_khr_spirv_no_integer_wrap_decoration_fadd_uint.spvasm64]: the entry point names also need updating to reflect what is being tested

File: test_conformance/spirv_new/test_cl_khr_spirv_no_integer_wrap_decoration.cpp [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    SPIRV_CHECK_ERROR(err, "Failed to create rhs buffer");
+
+    err = clEnqueueWriteBuffer(queue, rhs, CL_TRUE, 0, bytes, &h_rhs[0], 0, NULL, NULL);
+    SPIRV_CHECK_ERROR(err, "Failed to copy to rhs buffer");
+
+    std::string kernelStr;
+
+    {
+        std::stringstream kernelStream;
+        kernelStream << "#define spirv_fadd(a, b) (a) + (b)               \n";
```
Comment (aarongreig) [inline test_conformance/spirv_new/test_cl_khr_spirv_no_integer_wrap_decoration.cpp]: similar to my comments on the SPIR-V files can you update the `fmath` related names in this file to reflect what is being tested

File: test_conformance/spirv_new/test_cl_khr_spirv_no_integer_wrap_decoration.cpp [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        kernelStream << "    int id = get_global_id(0);                   \n";
+        kernelStream << "    out[id] = FUNC(lhs[id], rhs[id]);            \n";
+        kernelStream << "}                                                \n";
+        kernelStr = kernelStream.str();
+    }
+
+    size_t kernelLen = kernelStr.size();
+    const char *kernelBuf = kernelStr.c_str();
+
+    const char *options = "";
```
Comment (aarongreig) [inline test_conformance/spirv_new/test_cl_khr_spirv_no_integer_wrap_decoration.cpp]: I can't see where this is used, what is it for?

File: test_conformance/spirv_new/spirv_txt/ext_cl_khr_spirv_no_integer_wrap_decoration_fadd_uint.spvasm64 (hunk 1 above)
Comment (gwawiork) [reply test_conformance/spirv_new/spirv_txt/ext_cl_khr_spirv_no_integer_wrap_decoration_fadd_uint.spvasm64]: f states for function. Then we have function add with type uint. Can you accept this interpretation ?

File: test_conformance/spirv_new/test_cl_khr_spirv_no_integer_wrap_decoration.cpp (hunk 3 above)
Comment (gwawiork) [reply test_conformance/spirv_new/test_cl_khr_spirv_no_integer_wrap_decoration.cpp]: f states for function

File: test_conformance/spirv_new/test_cl_khr_spirv_no_integer_wrap_decoration.cpp (hunk 4 above)
Comment (gwawiork) [reply test_conformance/spirv_new/test_cl_khr_spirv_no_integer_wrap_decoration.cpp]: ok this looks like redundant

File: test_conformance/spirv_new/spirv_txt/ext_cl_khr_spirv_no_integer_wrap_decoration_fadd_uint.spvasm64 (hunk 1 above)
Comment (aarongreig) [reply test_conformance/spirv_new/spirv_txt/ext_cl_khr_spirv_no_integer_wrap_decoration_fadd_uint.spvasm64]: Ah my mistake, I was thinking like the `OpFAdd` vs `OpIAdd` opcodes. This is fine.

**Comment by kpet** [comment]
@aarongreig Any more comments before @gwawiork  sends an update?

**Comment by aarongreig** [comment]
the redundant string is the only issue I can see, otherwise LGTM

**Pull request merged**